api/main.py

```
@@ -54,87 +90,241 @@ async def root():
+        df = pd.read_csv(io.StringIO(contents.decode("utf-8")))
+
+        required_columns = ["Date", "Time", "Transaction Type", "Phone Number", "Amount"]
         missing_columns = [col for col in required_columns if col not in df.columns]
         if missing_columns:
-            raise HTTPException(
-                status_code=400,
-                detail=f"Missing required columns: {missing_columns}"
-            )
-        
-        # Engineer features
+            raise HTTPException(status_code=400, detail=f"Missing columns: {missing_columns}")
```

Ruff (0.14.4)

Abstract `raise` to an inner function

(TRY301)